JSQMessagesDemo/DemoMessagesViewController.m

```diff
@@ -111,7 +115,17 @@ - (void)viewDidAppear:(BOOL)animated
     self.collectionView.collectionViewLayout.springinessEnabled = [NSUserDefaults springinessSetting];
 }
 
+- (void)preferredContentSizeChanged:(NSNotification *)notification
+{
+    [self.collectionView layoutSubviews];
```

Comment:
never call `layoutSubviews` directly. call `setNeedsLayout`.

we should also invalidate the `collectionViewLayout`.